### PR TITLE
Kartik

### PR DESCRIPTION
**Pull request by kartik1998**
Added remaining backend API's

### CODE REVIEW

**Comment by kartik1998** [comment]
Tested the APi's please check.

File: routes/api/bus.js [hunk 1]
```diff
@@ -1,3 +1,64 @@
...
+    try {
+      let user = await User.findOne({ email });
+
+      if (user) {
+        return res
+          .status(400)
+          .json({ errors: [{ msg: 'User already exists' }] });
+      }
+
+      user = new User({
```
Comment (YatinGupta777) [inline routes/api/bus.js:43]: This still needs work. Should I just merge and you'll continue working on this ?

Comment (YatinGupta777) [reply routes/api/bus.js:43]: Because we are adding bus and not user.

Comment (kartik1998) [reply routes/api/bus.js:43]: what need to be changed?

Comment (kartik1998) [reply routes/api/bus.js:43]: damn i have added completely wrong code. Sorry man let me correct it.

File: routes/api/bus.js [hunk 2]
```diff
@@ -1,3 +1,64 @@
...
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+
+    const { name, email, password } = req.body;
+
+    try {
+      let user = await User.findOne({ email });
```
Comment (YatinGupta777) [inline routes/api/bus.js:35]: We are adding Bus, why it is referring User Model.

File: routes/api/bus.js [hunk 3]
```diff
@@ -1,3 +1,64 @@
...
+const User = require('../../models/User');
+const Bus = require('../../models/Bus');
+
+// @route    POST /addBus
+// @desc     Add a bus
+// @access   Private
+router.post(
+  '/addBus',
+  [
+    check('name', 'Name is required').not().isEmpty(),
```
Comment (YatinGupta777) [inline routes/api/bus.js:19]: Arent we using adminID for new Bus ?

File: routes/api/bus.js (hunk 1 above)
Comment (YatinGupta777) [reply routes/api/bus.js:43]: Please check above comments. I think there is some misunderstanding here, this code will not add a bus.

File: routes/api/bus.js (hunk 3 above)
Comment (kartik1998) [reply routes/api/bus.js:19]: Yes you are right. I'll explain the issue when this challenge is complete.

Comment (kartik1998) [reply routes/api/bus.js:19]: I am correcting and adding

**Comment by kartik1998** [comment]
I think I have finally uploaded the correct files. Please check and inform.

**Comment by kartik1998** [comment]
Unable to run your code on my machine.

**Pull request merged**